File: CS/DXApplication14/SchedulerService.cs
using DevExpress.Mvvm.UI;
using DevExpress.Xpf.Scheduling;

namespace DXApplication14
{
    public interface ISchedulerService
    {
        void ShowEditAppointmentWindow();
    }
    public class SchedulerService : ServiceBase, ISchedulerService
    {
        public void ShowEditAppointmentWindow()
        {
            var scheduler = this.AssociatedObject as SchedulerControl;
            if (scheduler != null)
                scheduler.Commands.ShowAppointmentWindowCommand.Execute(null);
        }
    }
}

File: VB/DXApplication14/SchedulerService.vb
Imports DevExpress.Mvvm.UI
Imports DevExpress.Xpf.Scheduling

Namespace DXApplication14
	Public Interface ISchedulerService
		Sub ShowEditAppointmentWindow()
	End Interface
	Public Class SchedulerService
		Inherits ServiceBase
		Implements ISchedulerService

		Public Sub ShowEditAppointmentWindow() Implements ISchedulerService.ShowEditAppointmentWindow
			Dim scheduler = TryCast(Me.AssociatedObject, SchedulerControl)
			If scheduler IsNot Nothing Then
				scheduler.Commands.ShowAppointmentWindowCommand.Execute(Nothing)
			End If
		End Sub
	End Class
End Namespace
